### PR TITLE
Add spawning capability

### PR DESCRIPTION
**Pull request by mrmacete**

### CODE REVIEW

File: src/io_frida.c [hunk 1]
```diff
@@ -179,20 +180,52 @@ static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
...
+			eprintf ("Invalid process specifier\n");
+			goto error;
+		}
+		if (!*argv) {
+			eprintf ("Invalid arguments for spawning\n");
+			r_str_argv_free (argv);
+			goto error;
+		}
+
+		path = g_strdup (argv[0]);
```
Comment (oleavr) [inline src/io_frida.c]: This copy isn't necessary right?

Comment (mrmacete) [inline src/io_frida.c]: oops, right!

Comment (mrmacete) [inline src/io_frida.c]: yeah even the variable can be avoided :raised_hands:

File: src/io_frida.c [hunk 2]
```diff
@@ -179,20 +180,50 @@ static RIODesc *__open(RIO *io, const char *pathname, int rw, int mode) {
...
 
-	if (!resolve_process (rf->device, process_specifier, &pid)) {
+	if (!spawn && !resolve_process (rf->device, process_specifier, &pid)) {
 		goto error;
 	}
 
-	rf->pid = pid;
-	rf->session = frida_device_attach_sync (rf->device, pid, &error);
+	if (spawn) {
+		char **argv;
```
Comment (trufae) [inline src/io_frida.c]: if you use r2 coding style you can defineand assign the vars in the same line, no need to define them at the begining of the frame

Comment (oleavr) [inline src/io_frida.c]: +1, we should definitely follow the r2 coding style in r2frida.

**Comment by trufae** [comment]
also, update the r2frida version to create a new release with support for this.

**Comment by mrmacete** [comment]
what version do you mean? the agent's `package.json`?

**Pull request merged**